common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/session/DeviceSessionCtx.java

```diff
@@ -165,6 +168,7 @@ private void updateDeviceSessionConfiguration(DeviceProfile deviceProfile) {
             MqttDeviceProfileTransportConfiguration mqttConfig = (MqttDeviceProfileTransportConfiguration) transportConfiguration;
             TransportPayloadTypeConfiguration transportPayloadTypeConfiguration = mqttConfig.getTransportPayloadTypeConfiguration();
             payloadType = transportPayloadTypeConfiguration.getTransportPayloadType();
+            transportType = DeviceTransportType.MQTT;
```

Comment:
you may just use a boolean variable mqttTransportType and avoid execution of ```DeviceTransportType.MQTT.equals(transportType);``` every time